src/Matrix.jl

```diff
@@ -2687,6 +2687,25 @@ end
 #
 ###############################################################################
 
+"""
+    is_alternating(M::MatElem)
+
+Return whether the form corresponding to the matrix `M` is alternating,
+i.e. `M = -transpose(M)` and `M` has zeros on the diagonal.
+Return `false` if `M` is not a square matrix.
+"""
+function is_alternating(M::MatrixElem)
+  n = nrows(M)
+  n == ncols(M) || return false
+  for i in 1:n
+    is_zero_entry(B, i, i) || return false
+    for j in (i + 1):n
+      B[i, j] == -B[j, i] || return false
+    end
+  end
```

Comment:
This mixes `B` and `M`. But let's just simplify it:
```suggestion
  is_skew_symmetric(M) || return false
  for i in 1:nrows(M)
    is_zero_entry(M, i, i) || return false
  end
```